### PR TITLE
New integration test for sapling balance

### PR DESCRIPTION
**Pull request by juanky201271**
Steps:
- zingo-testutils (zingolib) : https://github.com/zingolabs/zingolib/pull/564.
- zingo-regchest : Update with the new version of `zingolib`, create a new docker image 007.
- zingo-mobile: Update with the new version of `zingolib` & `zingo-regchest`.

### CODE REVIEW

File: android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt [hunk 1]
```diff
@@ -102,7 +102,7 @@ class ExecuteAddressesFromSeed {
         val mapper = jacksonObjectMapper()
 
         val server = "http://10.0.2.2:20000"
-        val chainhint = "main"
+        val chainhint = "regtest"
```
Comment (Oscar-Pepper) [inline android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt]: these are set to main to return addresses etc. that look like real addresses. as they are not connected to any server (Offline tests)) i think it makes sense to return "real" outputs

Comment (juanky201271) [reply android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt]: oh, yes, I don't know why I did this... sorry. I'll fix this.

File: android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt [hunk 2]
```diff
@@ -356,3 +356,60 @@ class ExecuteSummariesFromSeed {
...
+        // 2. Received in sapling pool =     +250_000
+        // 3. Received in transparent pool = +250_000
+        // 4. Send - 100_000 + 10_000fee =   -110_000
+        // 5. SendToSelf orchard pool =       -10_000 (one item: Fee)
+        // 6. SendToSelf sapling pool =       -10_000 (one item: Fee)
+        // 7. SendToSelf transparent pool =   -10_000 (two items: SendToSelf & Fee)
+        // 8. Shielding transparent pool =    -10_000 (one item: Fee)
+        // 9. Upgrading sapling pool =        -10_000 (one item: Fee)
+        //
+        // orchard pool = 84_000
```
Comment (Oscar-Pepper) [inline android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt]: can you update the comments with the correct value here too?

File: android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt [hunk 3]
```diff
@@ -356,3 +356,60 @@ class ExecuteSummariesFromSeed {
...
+        // sapling pool = 0
+        // transparent =  0
+
+        var balanceJson = RustFFI.execute("balance", "")
+        System.out.println("\nBalance:")
+        System.out.println(balanceJson)
+        val balancePreSend: Balance = mapper.readValue(balanceJson)
```
Comment (Oscar-Pepper) [inline android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt]: if we are not sending in this test, maybe this should just be called 'balance'?

**Review by Oscar-Pepper — CHANGES_REQUESTED** [review]

File: android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt (hunk 2 above)
Comment (juanky201271) [reply android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt]: done!

File: android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt (hunk 3 above)
Comment (juanky201271) [reply android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt]: done!

**Review by Oscar-Pepper — APPROVED** [review]
love the mobile integration tests rolling in!

**Pull request merged**